src/patch.rs

```diff
@@ -0,0 +1,401 @@
+            if listener.control_val.is_some() || listener.control_checked.is_some() {
+                listener.attach_control(el_ws);
+            } else {
+                listener.attach(el_ws, mailbox.clone());
+            }
+        }
+    }
+    for child in el.children.iter_mut() {
+        if let Node::Element(child_el) = child {
+            attach_listeners(child_el, mailbox);
+        }
+    }
+}
+
+/// Recursively detach event-listeners. Run this before patching.
+pub(crate) fn detach_listeners<Ms>(el: &mut El<Ms>) {
+    if let Some(el_ws) = el.node_ws.as_ref() {
+        for listener in &mut el.listeners {
+            listener.detach(el_ws);
+        }
+    }
+    for child in &mut el.children {
+        if let Node::Element(child_el) = child {
+            detach_listeners(child_el);
+        }
+    }
+}
+
+/// We reattach all listeners, as with normal Els, since we have no
+/// way of diffing them.
+pub(crate) fn setup_window_listeners<Ms>(
+    window: &Window,
+    old: &mut Vec<Listener<Ms>>,
+    new: &mut Vec<Listener<Ms>>,
+    mailbox: &Mailbox<Ms>,
+) {
+    for listener in old {
+        listener.detach(window);
+    }
+
+    for listener in new {
+        listener.attach(window, mailbox.clone());
+    }
+}
+
+/// Remove a node from the vdom and web_sys DOM.
+pub(crate) fn remove_node<Ms>(node: &web_sys::Node, parent: &web_sys::Node, el_vdom: &mut El<Ms>) {
+    websys_bridge::remove_node(node, parent);
+
+    if let Some(unmount_actions) = &mut el_vdom.hooks.will_unmount {
+        (unmount_actions.actions)(node);
+        //                if let Some(message) = unmount_actions.message.clone() {
+        //                    app.update(message);
+        //                }
+    }
+}
+
+/// Set up controlled components: Input, Select, and `TextArea` elements must stay in sync with the
+/// model; don't let them get out of sync from typing or other events, which can occur if a change
+/// doesn't trigger a re-render, or if something else modifies them using a side effect.
+/// Handle controlled inputs: Ie force sync with the model.
+fn setup_input_listener<Ms>(el: &mut El<Ms>)
+where
+    Ms: 'static,
+{
+    if el.tag == dom_types::Tag::Input
+        || el.tag == dom_types::Tag::Select
+        || el.tag == dom_types::Tag::TextArea
+    {
+        let listener = if let Some(checked) = el.attrs.vals.get(&dom_types::At::Checked) {
+            let checked_bool = match checked.as_ref() {
+                "true" => true,
+                "false" => false,
+                _ => panic!("checked must be true or false."),
+            };
+            events::Listener::new_control_check(checked_bool)
+        } else if let Some(control_val) = el.attrs.vals.get(&dom_types::At::Value) {
+            events::Listener::new_control(control_val.to_string())
+        } else {
+            // If Value is not specified, force the field to be blank.
+            events::Listener::new_control("".to_string())
+        };
```

Comment:
It allows us to add attribute `checked` to the standard text input and then it ignores `value` - maybe panics for wrong combinations or ignoring non-compatible attributes like this would be more suitable.